chia/_tests/core/server/test_rate_limits.py

```diff
@@ -25,355 +26,432 @@
-
-        saw_disconnect = False
-        for i in range(300):
-            response = r.process_msg_and_check(tx_message, rl_v2, rl_v2)
-            if response is not None:
-                saw_disconnect = True
-        assert saw_disconnect
-
-        r = RateLimiter(incoming=True)
-        block_message = make_msg(ProtocolMessageTypes.respond_unfinished_block, bytes([1] * 1024 * 1024))
-        for i in range(10):
-            assert r.process_msg_and_check(block_message, rl_v2, rl_v2) is None
-
-        saw_disconnect = False
-        for i in range(40):
-            response = r.process_msg_and_check(block_message, rl_v2, rl_v2)
-            if response is not None:
-                saw_disconnect = True
-        assert saw_disconnect
-
-    @pytest.mark.anyio
-    async def test_non_tx_aggregate_limits(self):
-        # Frequency limits
-        r = RateLimiter(incoming=True)
-        message_1 = make_msg(ProtocolMessageTypes.coin_state_update, bytes([1] * 32))
-        message_2 = make_msg(ProtocolMessageTypes.request_blocks, bytes([1] * 64))
-        message_3 = make_msg(ProtocolMessageTypes.plot_sync_start, bytes([1] * 64))
-
-        for i in range(500):
-            assert r.process_msg_and_check(message_1, rl_v2, rl_v2) is None
-
-        for i in range(500):
-            assert r.process_msg_and_check(message_2, rl_v2, rl_v2) is None
-
-        saw_disconnect = False
-        for i in range(500):
-            response = r.process_msg_and_check(message_3, rl_v2, rl_v2)
-            if response is not None:
-                saw_disconnect = True
-        assert saw_disconnect
-
-        # Size limits
-        r = RateLimiter(incoming=True)
-        message_4 = make_msg(ProtocolMessageTypes.respond_proof_of_weight, bytes([1] * 49 * 1024 * 1024))
-        message_5 = make_msg(ProtocolMessageTypes.request_blocks, bytes([1] * 49 * 1024 * 1024))
-
-        for i in range(2):
-            assert r.process_msg_and_check(message_4, rl_v2, rl_v2) is None
-
-        saw_disconnect = False
-        for i in range(2):
-            response = r.process_msg_and_check(message_5, rl_v2, rl_v2)
-            if response is not None:
-                saw_disconnect = True
-        assert saw_disconnect
-
-    @pytest.mark.anyio
-    async def test_periodic_reset(self):
-        r = RateLimiter(True, 5)
-        tx_message = make_msg(ProtocolMessageTypes.respond_transaction, bytes([1] * 500 * 1024))
-        for i in range(10):
-            assert r.process_msg_and_check(tx_message, rl_v2, rl_v2) is None
-
-        saw_disconnect = False
-        for i in range(300):
-            response = r.process_msg_and_check(tx_message, rl_v2, rl_v2)
-            if response is not None:
-                saw_disconnect = True
-        assert saw_disconnect
-        assert r.process_msg_and_check(tx_message, rl_v2, rl_v2) is not None
-        await asyncio.sleep(6)
+class SimClock:
+    current_time = 1000.0
```

Comment:
let's dataclass for clarity.  need to add the type hint anyways.  `:]`

(will need an import added as well)
```suggestion
@dataclass
class SimClock:
    current_time: float = 1000.0
```